### ISSUE TITLE
Signtool needs to understand spaces in paths inside VSIX extensions

### ISSUE THREAD

**Issue by MattGal** [dotnet/arcade#6242]
It seems while unpacking, caching, and checking for existing signing Signtool has problems if the paths inside the zip contain spaces:

Original failed run: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4093525&view=logs&j=4032467e-dc57-5727-0bde-0b3f32a19231&t=23a0f72a-24d0-5b93-813a-ea29ad9f7217

Deleted spaces: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4094069&view=logs&j=4032467e-dc57-5727-0bde-0b3f32a19231&t=23a0f72a-24d0-5b93-813a-ea29ad9f7217

Log
```
Extracting file 'Microsoft.VisualStudio.CodeCoverage.Shim.dll' from 'C:\A\1\14\s\artifacts\VSSetup\Release\Insertion\Microsoft.VisualStudio.TestTools.DynamicCodeCoverage.vsix' to 'C:\A\1\14\s\artifacts\tmp\Release\ContainerSigning\11\Contents/Team%20Tools/Dynamic%20Code%20Coverage%20Tools/coreclr/Microsoft.VisualStudio.CodeCoverage.Shim.dll'.
Tracking file 'C:\A\1\14\s\artifacts\tmp\Release\ContainerSigning\11\Contents/Team%20Tools/Dynamic%20Code%20Coverage%20Tools/coreclr/Microsoft.VisualStudio.CodeCoverage.Shim.dll' isNested=True
Caching file Microsoft.VisualStudio.CodeCoverage.Shim.dll EB72BDE17FBA05935A5B4BD5B9CF3CC39FAB2BA2D2C47D1380415E2007916DFF
Removing public sign: 'Microsoft.VisualStudio.CodeCoverage.Shim.dll'
C:\A\1\14\a\MicroBuild\Plugins\MicroBuild.Plugins.Signing.1.1.307\build\MicroBuild.Plugins.Signing.targets(43,5): Cannot StrongName sign the file at C:\A\1\14\s\artifacts\tmp\Release\ContainerSigning\11\Contents\Team Tools\Dynamic Code Coverage Tools\coreclr\Microsoft.VisualStudio.CodeCoverage.Shim.dll because it does not have a public key [C:\A\1\14\s\artifacts\tmp\Release\Signing\Round0.proj]
```

**Comment by ChadNedzlek** [comment]
It seems like whatever is storing them with the "%20" isn't correct here.  I'm not familiar with what parts that are though. :-)  Treating a URL blindly as a filename without dealing with escaping is bound to cause issues.

**Comment by MattGal** [comment]
Sure, but given it's inside our code, if we determine that this VSIX was malformed, erroring out instead of trying to do stuff with a path that is likely just the path up til the first space.

**Comment by ChadNedzlek** [comment]
Indeed. I feel like whatever put the file on disk is the "wrong" piece of code here, but it's probably a nasty 50/50 split, where half the code is URL escaping it, and the rest isn't.  So if we had to pick a direction to prefer, it seems like having the file on disk not have "%20" would be best, since that accurately represents the actual file.

**Comment by riarenas** [comment]
This is a duplicate of https://github.com/dotnet/arcade/issues/6239. We'll see if the customer is still blocked and see how we should deal with this.

**Comment by jakubch1** [comment]
We are still blocked on this. We want to replace existing Dynamic Code Coverage components and build them outside of VS repo. We want to keep existing files in same locations (Team Tools/Dynamic Code Coverage) as it contains tool CodeCoverage.exe that is used by external users. Signing is failing in both cases whatever folder I will specify in swr file:
folder "InstallDir:\Team Tools\Dynamic Code Coverage Tools"
folder "InstallDir:\Team%20Tools\Dynamic%20Code%20Coverage%20Tools"

It works only without spaces but changing location of CodeCoverage.exe is something that we don't want to do.
